control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -565,29 +1008,23 @@ func TestReconcileCreatePod(t *testing.T) {
 			podName: "foo",
 			k8sObjects: func() []runtime.Object {
 				pod := createPod("foo", "10.0.0.1", "foo", true, true)
+				podBytes, _ := json.Marshal(pod)
+				pod.Annotations[constants.AnnotationOriginalPod] = string(podBytes)
+
+				// Fake the probe changes that would be added by the mesh webhook
+				pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Port = intstr.FromInt(20300)
+				pod.Spec.Containers[0].LivenessProbe.HTTPGet.Port = intstr.FromInt(20400)
+				pod.Spec.Containers[0].StartupProbe.HTTPGet.Port = intstr.FromInt(20500)
```

Comment:
~ Here and elsewhere: Should we `TODO` a follow-up to use a shared helper so these tests don't drift from the hook impl? If not, maybe a local helper for reuse?

Reply:
👍 I added a local helper to simplify.